### PR TITLE
Split

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #671 

### Summary of Changes

* Updates split box
* Adds max / min widths to our panels to avoid bad behavior
* Adds overflow hidden to some elements

### Screenshots/Videos (OPTIONAL)

Edge Cases:
* can collapse panel to close to 0
* can expand the left panel without pushing the right sidebar off the window
* can expand the right panel without causing the app to add a scrollbar

### CODE REVIEW

File: public/js/components/SplitBox.js [hunk 1]
```diff
@@ -1,60 +1,219 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this file,
- * You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-const React = require("react");
+const React = require("devtools/client/shared/vendor/react");
 const ReactDOM = require("react-dom");
```
Comment (jasonLaster) [inline public/js/components/SplitBox.js]: had to use the relative path because the absolute path wouldn't resolve

**Comment by jlongster** [comment]
Shouldn't the SplitBox component now live in `lib/devtools`?

**Comment by jasonLaster** [comment]
Yup, we can move it to lib/devtools and land those fixes in m-c

File: public/js/components/Editor.css [hunk 2]
```diff
@@ -76,7 +76,7 @@
 }
 
 .welcomebox {
-  width: 100%;
+  width: calc(100% - 1px);
```
Comment (jasonLaster) [inline public/js/components/Editor.css]: this is a super strange issue w/ how the splitter handle is bigger than the splitter line.

**Pull request merged**